### ISSUE TITLE
Fido Ported to Unix

### ISSUE THREAD

**Issue by ElliotKillick** [pbatard/Fido#73]
Hi Pete,

Great job reverse engineering Microsoft's API! I was looking to do the same thing some time ago but didn't attempt due to assuming that it would require Selenium (with JS execution) or something along those lines.

Fido has recently been ported over to Unix using just POSIX sh, coreutils, and curl for the purpose of my open source Qubes OS project:
https://github.com/ElliotKillick/qvm-create-windows-qube/blob/master/windows/isos/mido.sh

I saw you recently dropped PowerShell Core support meaning Fido can no longer be run on Linux, macOS, or any system but Windows now. To get the non-Windows users out of your hair for good, it could be a good idea to add a link to Mido in your README? It's up to you but I would greatly appreciate it as well.

Thank you!

**Comment by alexsch01** [comment]
To be fair, I believe Fido still supports PowerShell Core running on Windows

**Comment by pbatard** [comment]
Yes, but OP is right that any platform other than Windows will produce an error, [by design](https://github.com/pbatard/Fido/blob/master/Fido.ps1#L64-L76).

I do like what the mido script does, and I don't exactly mind advertising for it, but I see that it uses archive.org to link to Windows 7 ISOs, and I'm not too sure where that falls in terms of copyright, because one could argue that, if someone uploads content that is infringing copyright, which archive.org then mirrors, and the original is taken down, you may end up with archive.org still serving the infringing content, so having something hosted on archive.org still does not mean it is okay to serve it, which is why I make sure to only serve content that is actively provided my Microsoft and Microsoft only in Fido...

**Comment by ElliotKillick** [comment]
Haven't really considered the legal aspect of archive.org redistributing software but I guess it could be legally troublesome. Recently, they [lost a court case](https://www.npr.org/2023/03/26/1166101459/internet-archive-lawsuit-books-library-publishers) against book publishers due to a service they provided where people could scan copyrighted books and upload them for anyone to read (archive.org says they're appealing the decision). However, (and I'm by no means a lawyer) I think the argument is very different with software because a vendor could suddenly stop providing some code that you need to operate (even while Windows 7 was in support, I saw this happen with some critical MSUs that MS stopped providing but could still be found on archive.org). Also, I believe it's different because this download has always been publicly available for indexing by bots until MS pulled it (unlike books where someone would have had to buy and scan it first). We will probably see a case like this get tried in the legal system at one point (and I hope archive.org wins) but until a verdict is reached I think it's fine to keep the Windows 7 ISO download link. Thoughts?

**Comment by ElliotKillick** [comment]
To be clear: I agree with the publishers in the above case, what archive.org did was a blatant copyright violation and I'm surprised they thought they could get away with that.

> if someone uploads content that is infringing copyright

Right. And by extension, I would also agree that archive.org pages like these:
https://archive.org/download/digital_river
https://archive.org/details/windows-10-20-h-2_202203

are almost certainly in strict violation of copyright because someone did have to upload them to be hosted on archive.org (again, I'm not a lawyer).

However, the Windows 7 link in Mido is from **web**.archive.org and looks like this:
https://web.archive.org/web/20221228154140/https://download.microsoft.com/download/5/1/9/5195A765-3A41-4A72-87D8-200D897CBE21/7601.24214.180801-1700.win7sp1_ldr_escrow_CLIENT_ULTIMATE_x64FRE_en-us.iso

It was directly scraped from the existing Microsoft link by a bot (no human upload) just as any other data would be. Mass data scraping of all kinds by bots is a very common practice these days especially for big tech companies like Google (they scape and then redistribute copyrighted content through Google Web Cache) and Microsoft too (recently as training data for their public Bing/GitHub Copilot AIs). It's also worth noting that web.archive.org sources a lot of its crawl data containing copyrighted content from [Amazon Alexa](https://archive.org/details/alexacrawls) redistributing it to them ever since 1996. So, it does indeed appear that this latter form is a lot more legally permissible and huges companies have been benefiting immensely by doing it for a long time...

**Comment by ave9858** [comment]
I added basic Linux support in my fork at https://github.com/ave9858/Fido, CLI mode only and haven't done much testing, I plan to make some more changes over time. Not going to attempt Windows 7 support using that method because potential legal problems, that ISO was poorly made and pulled by Microsoft for a reason, and Windows 7 is EOL and really shouldn't be used.

**Comment by 59de44955ebd** [comment]
Concerning this "infringing copyright when using archive.org" issue, I recently just for fun created a "clone" (in no way affiliated with Fido) that does only this, download Windows .isos from archive.org:
https://github.com/59de44955ebd/FidoArch
